src/Slices/CompileReports/UI/Page.tsx

```diff
@@ -35,55 +33,67 @@ export const Page: React.FC = () => {
     default: { name: "requested", order: "desc" },
     route: "CompileReports",
   });
-  const [data, retry] = queryResolver.useContinuous<"GetCompileReports">({
-    kind: "GetCompileReports",
+  const { data, refetch, isSuccess, isError, error } = useGetCompileReports({
     filter,
     sort,
     pageSize,
     currentPage,
-  });
+  }).useContinuous();
 
   //when sorting is triggered, reset the current page
   useEffect(() => {
     setCurrentPage({ kind: "CurrentPage", value: "" });
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [sort.order]);
 
+  if (isError) {
+    return (
+      <PageContainer pageTitle={words("compileReports.title")}>
+        <ErrorView
+          message={error.message}
+          retry={() => refetch()}
+          ariaLabel="CompileReportsView-Error"
+        />
+      </PageContainer>
+    );
+  }
+
+  if (isSuccess) {
+    return (
+      <PageContainer pageTitle={words("compileReports.title")}>
+        <CompileReportsTableControls
+          filter={filter}
+          setFilter={setFilter}
+          paginationWidget={
+            <PaginationWidget
+              data={data}
+              pageSize={pageSize}
+              setPageSize={setPageSize}
+              setCurrentPage={setCurrentPage}
+            />
+          }
+          afterRecompile={() => refetch()}
+        />
+        {data.data.length <= 0 ? (
```

Comment:
don't we typically do the select method on the GET queries to avoid data.data?

Reply:
queries with pagination have format of {data:[], links:{}, metadata:{}}